airflow/models/sqla_models.py

```diff
@@ -0,0 +1,13 @@
+"""Load SQLAlchemy models during package initialization"""
+
+# pylint: disable=unused-import
+
+# Must be loaded after loading DAG model.
+# noinspection PyUnresolvedReferences
+from airflow.jobs.base_job import BaseJob  # isort # isort:skip
+# noinspection PyUnresolvedReferences
+from airflow.jobs.scheduler_job import SchedulerJob  # isort # isort:skip
+# noinspection PyUnresolvedReferences
+from airflow.jobs.backfill_job import BackfillJob  # isort # isort:skip
+# noinspection PyUnresolvedReferences
+from airflow.jobs.local_task_job import LocalTaskJob  # isort # isort:skip
```

Comment:
The whole content of `airflow/models/__init__.py` should be moved here in the next steps. And then in `airflow/utils/db.py` we should replace `from airflow.models import ...` to `from airflow.models.sqla_models`. In this way, we should be able to avoid a lot of cyclic imports.

After all it may be possible to get rid of this file always use explicit import like `from airflow.models.dag import DAG`. 

WDYT? @ashb @potiuk

Reply:
Fully agree. Same problem that we had with airflow.__init__.py

Reply:
Do we need them all in a module at all? or would a method somewhere of `load_all_database_models()` work? I'm not too familiar with how SQLA models work at run time.

Reply:
Yes. That was the plan - to move them to a separate "load_all_modles() method". The models register themselves in "schema" when they are imported :(. I hate implicit behavior..

Reply:
Just be glad we're not using Rails :)

Reply:
What is the effect if we don't import all the models at runtime?

Reply:
From a quick re-read of the SQLA docs this would only affect if we tried to make SQLA issue table-creation commands?

Reply:
The main problem is that resetdb() willl not drop the tables.

Reply:
So eventually - the "load_all_models" should only be run in resetdb() case.

Reply:
I am wondering if we use `load_all_models` using importlib and strings...

Reply:
In the end the additional file was unecessary. I just imported `BaseJob` in the `models/db.py`